Tests/SwiftMemcacheTests/UnitTest/MemcachedResponseDecoderTests.swift

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOEmbedded
+@testable import SwiftMemcache
+import XCTest
+
+final class MemcachedResponseDecoderTests: XCTestCase {
+    var decoder: MemcachedResponseDecoder!
+    var channel: EmbeddedChannel!
+
+    override func setUp() {
+        super.setUp()
+        self.decoder = MemcachedResponseDecoder()
+        self.channel = EmbeddedChannel(handler: ByteToMessageHandler(self.decoder))
+    }
+
+    override func tearDown() {
+        XCTAssertNoThrow(try self.channel.finish())
+    }
+
+    func testDecodeSetResponse(returnCode: [UInt8], expectedReturnCode: MemcachedResponse.ReturnCode) throws {
+        // Prepare a response buffer with a response code
+        var buffer = ByteBufferAllocator().buffer(capacity: 8)
+        buffer.writeBytes(returnCode)
+        buffer.writeBytes([UInt8.carriageReturn, UInt8.newline])
```

Comment:
Can we create a helper method called `makeMemcachedResponseByteBuffer` that takes a `MemcachedResponse` and encodes it for us. We will have a bunch more tests down the line and this will help us write them easily